weave-js/src/components/PagePanelComponents/Home/Browse3/pages/PlaygroundPage/PlaygroundChat/InferenceBanner.tsx

```diff
@@ -0,0 +1,63 @@
+import React from 'react';
+
+import {Alert} from '../../../../../../Alert';
+import {AccountPlanType} from '../../../inference/types';
+import {useInferenceContext} from '../../../inference/util';
+
+export const InferenceBanner = () => {
+  const inferenceContext = useInferenceContext();
+  const {billing} = inferenceContext;
+  if (!billing) return null;
+  let {
+    accountPlanType,
+    usage,
+    inferenceSafetyLimit,
+    billingPeriodEnd,
+    getFormattedBillingDate,
+  } = billing;
+
+  let exceededLimitBoldStatusMsg: string | undefined;
+  let exceededLimitStatusMsg: string | undefined;
+
+  switch (accountPlanType) {
+    case AccountPlanType.Free:
+      if (inferenceSafetyLimit * 0.9 <= usage && usage < inferenceSafetyLimit) {
```

Comment:
Move this outside the switch maybe? Looks like all cases have the same condition.